include/operations/blas2_trees.h

```diff
@@ -347,6 +347,44 @@ make_tbmv(lhs_t &lhs_, matrix_t &matrix_, typename vector_t::index_t k_,
               is_unit>(lhs_, matrix_, k_, vector_);
 }
 
+/**
+ * @struct Trsv
+ * @brief Tree node representing a triangular matrix_ vector_
+ * multiplication.
+ */
+template <typename lhs_t, typename matrix_t, typename vector_t,
+          uint32_t local_range, bool is_upper, bool is_transposed, bool is_unit>
+struct Trsv {
+  using value_t = typename vector_t::value_t;
+  using index_t = typename vector_t::index_t;
+
+  lhs_t lhs_;
+  matrix_t matrix_;
+  index_t blk_id_;
+  vector_t vector_;
+
+  Trsv(lhs_t &_l, matrix_t &_matrix, index_t &_k, vector_t &_vector);
+  index_t get_size() const;
+  bool valid_thread(cl::sycl::nd_item<1> ndItem) const;
+  value_t eval(index_t i);
+  value_t eval(cl::sycl::nd_item<1> ndItem);
+  template <typename local_memory_t>
+  value_t eval(local_memory_t local_mem, cl::sycl::nd_item<1> ndItem);
```

Comment:
You don't really need multiple `eval` functions. I would say let's keep the one that we have implemented in the `src/` directory and remove the rest. Also, I heard it from someone (Mehdi/Duncan) that having a return type for the kernel may not be the best practice. cc: @mehdi-goli

Reply:
I removed unnecessary methods. I can remove the return type as well since it is not necessary. I introduced it to be consistent with the other function.